spec/cypress/integration/admin/LeadProviderUserManagement.feature

```diff
@@ -0,0 +1,73 @@
+Feature: Admin user modifying lead provider users
+  Admin users should be able to create and delete lead provider users
+
+  Background:
+    Given I am logged in as an "admin"
+    And scenario "admin/suppliers" has been run
```

Comment:
We could convert data setup scenarios into concrete steps using Factory bot support

Reply:
This scenario is reused over multiple feature specs, I'd prefer standardised testing data across scenarios than a more readable specification with unstandardised data

Reply:
Let's imagine a scenario where you share that feature file with the tester/product owner for a shared context - all the other steps could be easily conceptualized and transparent - easy to read. However, the scenarios obfuscate that data and as such shared meaning and communication have barriers. Cucumber is not only for testing but also for documentation of the system.

Reply:
```
delivery_partner = FactoryBot.create(:delivery_partner, name: "Delivery Partner 1")
cohort = FactoryBot.create(:cohort, start_year: 2021)
lead_provider = FactoryBot.create(:lead_provider, name: "Lead Provider 1", cohorts: [cohort])
ProviderRelationship.create!(delivery_partner: delivery_partner, lead_provider: lead_provider, cohort: cohort)
```

How would you write this as a cucumber spec?

Reply:
Given a provider relationship is being created between DP: "Delivery Partner 1" and LP: "Lead Provider 1" for Cohort: "2021" - there would be a couple of tweaks to create Factory for that so that it could be called from cypress.

Reply:
It isn't just creating the relationship, though, it's creating the delivery partner and lead provider too

Reply:
I think running a scenario is a black box for non-dev person. Maybe we should think about making it more english, but can we make it in a separate PR? I wanna get some changes Callum has here to prettify accessible autocomplete specs.

Reply:
Have already merged!

I agree to an extent - and for creating a user, it's probably better to have that in english in the spec. However, when creating a delivery partner and lead provider and relationship between them, I don't think there _is_ a way to write that in plain english